org.eclipse.lsp4e/src/org/eclipse/lsp4e/DocumentContentSynchronizer.java

```diff
@@ -270,7 +237,7 @@ public void documentAboutToBeSaved() {
 
 
 		try {
-			List<TextEdit> edits = executeOnCurrentVersionAsync(ls -> ls.getTextDocumentService().willSaveWaitUntil(params))
+			List<TextEdit> edits = languageServerWrapper.executeImpl(ls -> ls.getTextDocumentService().willSaveWaitUntil(params))
```

Comment:
why not calling `languageServerWrapper.execute`? The javadoc of `executeImpl` says
 "Internal hook for the LSPexecutor implementations", which does not hold for this case.

Reply:
Well, `DocumentContentSynchronizer` is 'trusted code' for the purposes of using the internal API. The difference between the internal and external is just the extra `.andThenAsync(Function.identity())` designed to make sure the general public doesn't tie up listener threads by composing lots of extra work to the returned future. Since we're just doing a get (and indeed a timeout!) here there is no need for the extra async, and indeed we want the result as quickly as possible without any potential scheduling delay in delivering it via a pool thread.